src/core/syntax-tree/lexicon.ts

```diff
@@ -33,22 +33,22 @@ const isPrime = (c: number) => {
 
 const operand = {
     resolve: (target: any, tree: ISyntaxTree) => {
-        if (/^('.*')|(".*")$/.test(tree.value)) {
+        if (/^(('([^'\\]|\\'|\\)+')|("([^"\\]|\\"|\\)+")|(`([^`\\]|\\`|\\)+`))$/.test(tree.value)) {
```

Comment:
Is `'([^'\\]|\\'|\\)+'` different from `'([^']|\\')+'`?

Reply:
oh actually, it's important to NOT match `'blah\'` (missing the real closing quote) which seems like both of these would match?
So perhaps `'([^'\\]|\\.)+'`?

Reply:
You're right. Will add some tests for it too.